spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestParquetMergeOnRead.java

```diff
@@ -0,0 +1,196 @@
+import org.apache.iceberg.parquet.Parquet;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.spark.SparkCatalogConfig;
+import org.apache.iceberg.spark.SparkSchemaUtil;
+import org.apache.iceberg.spark.data.RandomData;
+import org.apache.iceberg.spark.data.SparkParquetWriters;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+public class TestParquetMergeOnRead extends SparkExtensionsTestBase {
+
+  private final boolean vectorized;
+
+  public TestParquetMergeOnRead(
+      String catalogName, String implementation, Map<String, String> config, boolean vectorized) {
+    super(catalogName, implementation, config);
+    this.vectorized = vectorized;
+  }
+
+  @Parameterized.Parameters(
+      name = "catalogName = {0}, implementation = {1}, config = {2}, vectorized = {3}")
+  public static Object[][] parameters() {
+    return new Object[][] {
+      {
+        SparkCatalogConfig.SPARK.catalogName(),
+        SparkCatalogConfig.SPARK.implementation(),
+        SparkCatalogConfig.SPARK.properties(),
+        false
+      },
+      {
+        SparkCatalogConfig.SPARK.catalogName(),
+        SparkCatalogConfig.SPARK.implementation(),
+        SparkCatalogConfig.SPARK.properties(),
+        true
+      }
+    };
+  }
+
+  @BeforeClass
+  public static void startMetastoreAndSpark() {
+    metastore = new TestHiveMetastore();
+    metastore.start();
+    HiveConf hiveConf = metastore.hiveConf();
+
+    spark =
+        SparkSession.builder()
+            .master("local[2]")
+            .config("spark.sql.extensions", IcebergSparkSessionExtensions.class.getName())
+            .config("spark.hadoop." + METASTOREURIS.varname, hiveConf.get(METASTOREURIS.varname))
+            .enableHiveSupport()
+            .getOrCreate();
+
+    catalog =
+        (HiveCatalog)
+            CatalogUtil.loadCatalog(
+                HiveCatalog.class.getName(), "hive", ImmutableMap.of(), hiveConf);
+
+    try {
+      catalog.createNamespace(Namespace.of("default"));
+    } catch (AlreadyExistsException ignored) {
+      // The default namespace already exists. Ignore the error.
+    }
+  }
+
+  @AfterClass
+  public static void stopMetastoreAndSpark() throws Exception {
+    catalog = null;
+    metastore.stop();
+    metastore = null;
+    spark.stop();
+    spark = null;
+  }
+
+  @After
+  public void dropTable() {
+    spark.sql("DROP TABLE IF EXISTS default.test_iceberg");
+  }
+
+  @Test
+  public void testReadOfDataFileWithMultipleSplits() throws IOException {
```

Comment:
I think the purpose is to test spark row-level delete read in case of multiple row groups. As long as we create multiple row groups in parquet file, we should be good.  We can use the same way in this class to make sure of generating row groups, https://github.com/apache/iceberg/blob/bac84bcc580a392955eb520b9e3c316a16af62b7/spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkParquetReadMetadataColumns.java#L133-L133

Reply:
I agree that the bug simply depends on having multiple row groups in the data file that has applicable positional deletes.

Reply:
Thanks for the pointer.